source/MRViewer/MRSurfaceManipulationWidget.cpp

```diff
@@ -772,8 +777,14 @@ void SurfaceManipulationWidget::updateVizualizeSelection_( const ObjAndPick& obj
             }
             pOnFace = PointOnFace{ objAndPick.second.face, mesh.points[vert] };
         }
-        visualizationDistanceMap_ = computeSpaceDistances( mesh, pOnFace, settings_.radius );
-        visualizationRegion_ = findNeighborVerts( mesh, pOnFace, settings_.radius );
+
+        std::pair<VertScalars, VertBitSet> distancesAndRegion;
+        if ( editOnlyVisibleSurface_ )
+            distancesAndRegion = findSpaceDistancesAndRegionVisible( mesh, pOnFace, settings_.radius, mesh.normal( mesh.toTriPoint( pOnFace ) ) );
+        else
+            distancesAndRegion = findSpaceDistancesAndRegion( mesh, pOnFace, settings_.radius );
+        visualizationDistanceMap_ = std::move( distancesAndRegion.first );
+        visualizationRegion_ = std::move( distancesAndRegion.second );
```

Comment:
move it into separate function inside this class not to duplicate

Reply:
fixed